src/sentry/models/project.py

```diff
@@ -176,9 +177,12 @@ def save(self, *args, **kwargs):
                     reserved=RESERVED_PROJECT_SLUGS,
                     max_length=50,
                 )
-            super().save(*args, **kwargs)
-        else:
-            super().save(*args, **kwargs)
+
+        snowflake_redis_key = "project_snowflake_key"
+        self.save_with_snowflake_id(
+            snowflake_redis_key, lambda: super(Project, self).save(*args, **kwargs)
+        )
```

Comment:
@maxiuyuan since this is a dangerous migration, it might be better to separate out these changes into a follow up PR.

Reply:
https://github.com/getsentry/sentry/pull/37815 done!